src/formatDate.js

```diff
@@ -8,7 +8,35 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const separatorFrom = fromFormat.pop();
+  const separatorTo = toFormat.pop();
+  const dateArr = date.split(separatorFrom);
+  const resultArr = [];
+
+  for (let i = 0; i < toFormat.length; i++) {
+    const format = toFormat[i];
+    const index = fromFormat.indexOf(format);
+
+    if (index === -1) {
+      if (format === 'YYYY') {
+        const fromIndex = fromFormat.indexOf('YY');
+        const year = dateArr[fromIndex];
+
+        resultArr.push(year < 30 ? '20' + year : '19' + year);
```

Comment:
The comparison `year < 30` might not work as expected because `year` is a string. You should convert it to a number before comparing. Also, consider handling edge cases where `year` is not a valid number.